sunpy/map/maputils.py

```diff
@@ -124,6 +124,17 @@ def sample_at_coords(smap, coordinates):
     return smap.data[smap.wcs.world_to_array_index(coordinates)]
 
 
+def _edge_coordinates(smap):
+    edges = map_edges(smap)
+    # Calculate all the edge pixels
+    edge_pixels = np.concatenate(edges)
+    # Need this line for compatability with old versions of astropy, where
+    # edge_pixels is not a full on Quantity. This can be removed in the future
+    edge_pixels = edge_pixels.value * edges[0].unit
```

Comment:
(I can't check myself right now...)  What is `edge_pixels` if it's not a `Quantity`?  What happens if you instead used:
```python
edge_pixels = np.concatenate(edges) << edges[0].unit
```

Reply:
I'm not sure on old versions of astropy, but this was what I had to do to get the oldtestdeps build to pass. Your suggestion seems to work fine (on new versions of astropy), but I've never seen `<<` before (the internet says it is a bitwise shift operator?) and I think the current version is a bit clearer

Reply:
astropy overloaded `<<` to mean "convert to quantity without copying the array" iirc.

Reply:
https://docs.astropy.org/en/stable/units/quantity.html#astropy-units-quantity-no-copy

Reply:
It doesn't have to be the insertion/left-shift operator (`<<`), which is admittedly unusual to those who aren't familiar with C++ streams or with bit operations.  It could instead be the `Quantity()` initializer:
```python
edge_pixels = Quantity(np.concatenate(edges), edges[0].unit, copy=False)
```
I don't like your current code pattern because it utilizes specific aspects of the API rather than abstractly requesting what you want, it creates a copy when one isn't needed, and it performs an operation even when `edge_pixels` is already a `Quantity`.

Reply:
I'm still not clear what `edge_pixels` is if it's not a `Quantity`; why does `edge_pixels.value` work in that case?

Reply:
Some investigation; the problem occurs with our oldest supported version of numpy (`1.16.6`), but is not an issue for `numpy>=1.17`. The `<<` suggestion doesn't work (where the current PR does), failing with
```
Traceback (most recent call last):
  File "test.py", line 10, in <module>
    coords = sunpy.map.is_all_on_disk(m)
  File "/Users/dstansby/github/sunpy/sunpy/map/maputils.py", line 285, in is_all_on_disk
    edge_of_world = _edge_coordinates(smap)
  File "/Users/dstansby/github/sunpy/sunpy/map/maputils.py", line 133, in _edge_coordinates
    edge_pixels = np.concatenate(edges) << edges[0].unit
  File "/Users/dstansby/miniconda3/envs/dev/lib/python3.8/site-packages/astropy/units/quantity.py", line 887, in __lshift__
    return self.__class__(self, other, copy=False, subok=True)
  File "/Users/dstansby/miniconda3/envs/dev/lib/python3.8/site-packages/astropy/units/quantity.py", line 300, in __new__
    value = value.to(unit)
  File "/Users/dstansby/miniconda3/envs/dev/lib/python3.8/site-packages/astropy/units/quantity.py", line 689, in to
    return self._new_view(self._to_value(unit, equivalencies), unit)
  File "/Users/dstansby/miniconda3/envs/dev/lib/python3.8/site-packages/astropy/units/quantity.py", line 660, in _to_value
    return self.unit.to(unit, self.view(np.ndarray),
AttributeError: 'NoneType' object has no attribute 'to'
```
`edge_pixels` is of type `<class 'astropy.units.quantity.Quantity'>`, but for some reason does not have a unit attribute properly propagated to it during the `np.concatenate` call.

Reply:
I've noticed that this can happen with some numpy and scipy functions. They will return a `Quantity` but for some reason the units will be stripped off and it has never been completely clear to me why. Maybe at this stage you should explicitly strip units off of `edges` before passing to `np.concatenate` so that it is completely clear that the `edge_pixels` returned in L30 is a numpy array and then just remove the `.value` in L33.